presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -754,6 +770,25 @@ private PlanBuilder appendScalarSubqueryJoin(PlanBuilder builder, SubqueryExpres
         }
     }
 
+    private ProjectNode coercionScalarSubquery(PlanNode node, SubqueryExpression scalarSubquery, RelationPlan relationPlan)
+    {
+        TranslationMap translations = new TranslationMap(relationPlan, analysis);
```

Comment:
relationPlan should be a RelationPlan based on enforceSingleNode created by the caller. This on is based on scalarSubquery. So, you shouldn't have to pass both `node` and `relationPlan`, since `node` would be `relationPlan.getRoot()`